engine/linksystem.go

```diff
@@ -52,14 +61,24 @@ func (e *Engine) mkLinkSystem() ipld.LinkSystem {
 				log.Errorf("Could not decode IPLD node for potential advertisement: %s", err)
 				return nil, err
 			}
+
 			// If this was an advertisement, then return it.
 			if isAdvertisement(n) {
+				if cidSchemaType != "" && cidSchemaType != ipnisync.CidSchemaAdvertisement {
+					log.Errorf("Retrieved advertisement when asked for %s", cidSchemaType)
+					return nil, errors.New("data does not match requested type")
+				}
 				log.Debugw("Retrieved advertisement from datastore", "cid", c, "size", len(val))
 				return bytes.NewBuffer(val), nil
 			}
 			log.Debugw("Retrieved non-advertisement object from datastore", "cid", c, "size", len(val))
 		}
 
+		if cidSchemaType != "" && cidSchemaType != ipnisync.CidSchemaEntryChunk {
+			log.Errorf("Asked for %s when expecting %s", cidSchemaType, ipnisync.CidSchemaEntryChunk)
+			return nil, errors.New("requested unexpected data type")
+		}
+
```

Comment:
lets log these but not return an error?

if we got the right data that matches the cid, we should continue, not error